tests/test_cliCmdQueryDatasets.py

```diff
@@ -0,0 +1,260 @@
+
+        # Create and register a DatasetType
+        dimensions = self.butler.registry.dimensions.extract(["instrument", "visit"])
+        datasetTypeName = "test_metric_comp"
+        self.datasetType = self._addDatasetType(datasetTypeName, dimensions, storageClass,
+                                                self.butler.registry)
+
+        # Add needed Dimensions
+        self.butler.registry.insertDimensionData("instrument", {"name": "DummyCamComp"})
+        self.butler.registry.insertDimensionData("physical_filter", {"instrument": "DummyCamComp",
+                                                                     "name": "d-r",
+                                                                     "band": "R"})
+        self.butler.registry.insertDimensionData("visit_system", {"instrument": "DummyCamComp",
+                                                                  "id": 1,
+                                                                  "name": "default"})
+        visit_start = astropy.time.Time("2020-01-01 08:00:00.123456789", scale="tai")
+        visit_end = astropy.time.Time("2020-01-01 08:00:36.66", scale="tai")
+        self.butler.registry.insertDimensionData("visit",
+                                                 {"instrument": "DummyCamComp", "id": 423,
+                                                  "name": "fourtwentythree", "physical_filter": "d-r",
+                                                  "visit_system": 1, "datetime_begin": visit_start,
+                                                  "datetime_end": visit_end})
+        self.butler.registry.insertDimensionData("visit", {"instrument": "DummyCamComp", "id": 424,
+                                                           "name": "fourtwentyfour", "physical_filter": "d-r",
+                                                           "visit_system": 1})
+        metric = self._makeExampleMetrics()
+        dataId = {"instrument": "DummyCamComp", "visit": 423}
+        ref = DatasetRef(self.datasetType, dataId, id=None)
+        self.butler.put(metric, ref)
+
+        metric = self._makeExampleMetrics()
+        dataId = {"instrument": "DummyCamComp", "visit": 424}
+        ref = DatasetRef(self.datasetType, dataId, id=None)
+        self.butler.put(metric, ref)
+
+    def tearDown(self):
+        if os.path.exists(self.root):
+            shutil.rmtree(self.root, ignore_errors=True)
+
+    def testShowURI(self):
+        """Test for expected output with show_uri True and False.
+
+        Check the entire result for completeness and basic structure. Other
+        tests should not need to do as much checking.
+        """
+        for showUri in True, False:
+            tables = self._queryDatasets(repo=self.butlerConfigFile, show_uri=showUri)
+
+            expectedNames = ["type", "run", "id", "instrument", "visit"]
+            if showUri:
+                expectedNames.append("URI")
+
+            for table in tables:
```

Comment:
Would it be clearer here to define the astropy.table.Table you expect to get and then use `assertEqual` directly? Rather than try to write code to compare all the rows and columns yourself?

Reply:
`assertEqual` (and variations) does not seem to work with astropy tables. I'm able to use `report_diff_values` with good effect though https://docs.astropy.org/en/stable/api/astropy.utils.diff.report_diff_values.html#astropy.utils.diff.report_diff_values
something like 
```
        diff = io.StringIO()
        self.assertEqual(len(tables), 1)
        self.assertTrue(report_diff_values(tables[0], expectedTable, fileobj=diff), msg=diff.getvalue())
```
does that seem ok?

Reply:
that works for me.